src/Core/Framework/App/Privileges/Privileges.php

```diff
@@ -0,0 +1,245 @@
+     */
+    public function acceptAllForApps(array $appIds, Context $context): void
+    {
+        $privileges = $this->fetchPrivileges($appIds);
+
+        foreach ($appIds as $appId) {
+            [$existingPrivileges, $requestedPrivileges] = $privileges[$appId];
+
+            $new = array_merge($existingPrivileges, $requestedPrivileges);
+
+            $this->writePrivileges($appId, $new, [], $context);
+        }
+    }
+
+    /**
+     * @param list<string> $appIds
+     */
+    public function revokeAllForApps(array $appIds, Context $context): void
+    {
+        foreach ($appIds as $appId) {
+            $this->writePrivileges($appId, [], [], $context);
+        }
+    }
+
+    /**
+     * Get the requested privileges for all active apps
+     *
+     * @return array<string, list<string>>
+     */
+    public function getRequestedPrivilegesForAllApps(): array
+    {
+        /** @var array<string, string> $privileges */
+        $privileges = $this->connection->fetchAllKeyValue(
+            <<<'SQL'
+                SELECT name, requested_privileges
+                FROM app
+                WHERE app.active = 1
+            SQL,
+        );
+
+        return $this->decodePrivileges($privileges);
+    }
+
+    /**
+     * Get the current privileges for the given list of apps
+     *
+     * @param list<string> $appIds
+     *
+     * @return array<string, list<string>>
+     */
+    public function getPrivileges(array $appIds = []): array
+    {
+        return array_map(
+            fn (array $privileges): array => $privileges[0],
+            $this->fetchPrivileges($appIds)
+        );
+    }
+
+    /**
+     *  Get the requested privileges for the given list of apps
+     *
+     * @param list<string> $appIds
+     *
+     * @return array<string, list<string>>
+     */
+    public function getRequestedPrivileges(array $appIds = []): array
+    {
+        /** @var array<string, string> $privileges */
+        $privileges = $this->connection->fetchAllKeyValue(
+            <<<'SQL'
+                SELECT LOWER(HEX(app.id)) AS app_id, requested_privileges
+                FROM app
+                WHERE id IN (:ids)
+            SQL,
+            ['ids' => Uuid::fromHexToBytesList($appIds)],
+            ['ids' => ArrayParameterType::STRING]
+        );
+
+        return $this->decodePrivileges($privileges);
+    }
+
+    /**
+     * @param list<string> $privileges
+     */
+    public function setPrivileges(string $appId, array $privileges, Context $context): void
+    {
+        $this->connection->executeStatement(
+            'UPDATE `acl_role` SET `privileges` = :privileges WHERE id = (SELECT acl_role_id FROM app WHERE id = :id)',
+            [
+                'privileges' => json_encode($privileges, \JSON_THROW_ON_ERROR),
+                'id' => Uuid::fromHexToBytes($appId),
+            ]
+        );
+
+        $this->eventDispatcher->dispatch(new AppPermissionsUpdated($appId, $privileges, $context));
```

Comment:
if I see that correctly, the context is only needed to have it in the event. do we need it at all? does the event need to be a `ShopwareEvent`? if not, could we drop the context from it?

Reply:
Not sure, I will look in to it if it really needs it

Reply:
I don't see anywhere where it's needed, what is the general consensus for when to use `ShopwareEvent` ? Eg the app events/webhooks use it

Reply:
We decided to keep the context, since ext developers should use context from event and not create default context, ref: https://shopware-ag.slack.com/archives/C080HM3C85R/p1742314580707629